app/src/Controller/UsersController.php

```diff
@@ -104,4 +107,75 @@ public function delete($id = null)
 
         return $this->redirect(['action' => 'index']);
     }
+
+    /**
+     * Callback method executed before each controller action.
+     *
+     * This method is used to configure the authentication component by adding actions that
+     * do not require user authentication. It prevents an infinite redirect loop issue when
+     * unauthenticated users try to access login and add actions.
+     *
+     * @param \Cake\Event\EventInterface $event The event object.
+     * @return void
+     */
+    public function beforeFilter(EventInterface $event)
+    {
+        parent::beforeFilter($event);
+        // Configure the login action to not require authentication, preventing
+        // the infinite redirect loop issue
+        $this->Authentication->addUnauthenticatedActions(['login', 'add']);
```

Comment:
Please refer [CakePHP tutorial](https://book.cakephp.org/4/en/tutorials-and-examples/cms/authentication.html#:~:text=We%20want%20all%20view%20and%20index%20pages%20accessible%20without%20logging%20in%20so%20we%E2%80%99ll%20add%20this%20specific%20configuration%20in%20AppController)

I think you've missed the configuration highlighted in the CakePHP tutorial. Please check.